nhost/service/manager.go

```diff
@@ -86,13 +95,162 @@ func (m *dockerComposeManager) SetGitBranch(gitBranch string) {
 	m.composeConfig = compose.NewConfig(m.nhostConfig, m.ports, m.env, gitBranch, m.projectName)
 }
 
+func (m *dockerComposeManager) checkPorts(ctx context.Context) error {
+	select {
+	case <-ctx.Done():
+		return nil
+	default:
+	}
+
+	type composeSvc struct {
+		Service    string `json:"Service"`
+		Publishers []struct {
+			PublishedPort uint32 `json:"PublishedPort"`
+		}
+	}
+
+	cmd, err := compose.WrapperCmd(ctx, []string{"ps", "--filter", "status=running", "--format", "json"}, m.composeConfig, nil)
+	if err != nil && ctx.Err() != context.Canceled {
+		m.status.Error("Failed to list running services")
+		m.l.WithError(err).Debug("Failed to list running services")
+		return err
+	}
+
+	m.setProcessToStartInItsOwnProcessGroup(cmd)
+	out, err := cmd.CombinedOutput()
+	if err != nil && ctx.Err() != context.Canceled {
+		m.status.Error("Failed to list running services")
+		m.l.WithError(err).Debug("Failed to list running services")
+		return err
+	}
+
+	var svcs []composeSvc
+	err = json.Unmarshal(out, &svcs)
+	if err != nil {
+		m.status.Error("Failed to list running services")
+		m.l.WithError(err).Debug("Failed to list running services")
+		return err
+	}
+
+	runningSvcs := make(map[string]composeSvc)
+
+	for _, svc := range svcs {
+		runningSvcs[svc.Service] = svc
+	}
+
+	randomizerFunc := func(port uint32, timeout time.Duration) (uint32, error) {
+		t := time.After(timeout)
+
+		for i := 0; i < 1024; i++ {
+			select {
+			case <-t:
+				return 0, fmt.Errorf("timed out on ports randomize")
+			default:
+				if util.PortAvailable(fmt.Sprint(port)) {
+					return port, nil
+				}
+
+				port++
+			}
+		}
+
+		return 0, fmt.Errorf("failed to find available port")
+	}
+
+	portPublishedForSvc := func(port uint32, svc composeSvc) bool {
+		for _, p := range svc.Publishers {
+			if p.PublishedPort == 0 {
```

Comment:
you could rewrite this more succintly as:

``` golang
switch p.PublishedPort {
case 0:
   continue
case port:
   return true
default:
   return false
}
```

If you really prefer if statements though:

```
if p.PublishedPort == 0 {
    continue
} else if p.PublishedPort == port {
   return true
}
return false
```